src/cargo/core/shell.rs

```diff
@@ -18,10 +19,16 @@ pub enum Verbosity {
     Quiet
 }
 
+#[derive(Clone, Copy, PartialEq)]
+pub enum ColorConfig {
+    Auto,
+    Always,
+    Never
+}
+
 #[derive(Clone, Copy)]
 pub struct ShellConfig {
-    pub color: bool,
-    pub verbosity: Verbosity,
```

Comment:
When I first started investigating the problem I discovered that both `Shell` and `MultiShell` had verbosity options. The option in `Shell` always used it's default value and was never changed. I removed the option on my path to the first implementation for this PR and the original commit ended up squashed. Let me know if you would like to see it returned to keep this PR cleaner.